tools/RepositoryManagement/DevHome.RepositoryManagement/Services/EnhanceRepositoryService.cs

```diff
@@ -0,0 +1,170 @@
+    public async Task<(string RepositoryLocation, Guid sourceControlClassId)> SelectRepositoryAndMakeItEnhanced()
+    {
+        var maybeRepositoryPath = await GetRepositoryLocationFromUser();
+        var sourceControlId = await MakeRepositoryEnhanced(maybeRepositoryPath);
+        return (maybeRepositoryPath, sourceControlId);
+    }
+
+    public async Task<Guid> MakeRepositoryEnhanced(string repositoryLocation)
+    {
+        _sourceControlRegistrar.AddRepositoryAlreadyOnMachine(repositoryLocation);
+        var sourceControlId = await AssignSourceControlToPath(repositoryLocation);
+        return sourceControlId;
+    }
+
+    public string GetLocalBranchName(string repositoryLocation)
+    {
+        try
+        {
+            var repository = new LibGit2Sharp.Repository(repositoryLocation);
+            return repository.Head.FriendlyName;
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Error when getting branch name");
+        }
+
+        return string.Empty;
+    }
+
+    public string GetRepositoryUrl(string repositoryLocation)
+    {
+        try
+        {
+            var repository = new LibGit2Sharp.Repository(repositoryLocation);
+            return repository.Network.Remotes.First().Url;
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Error when getting the repositoryUrl");
+        }
+
+        return string.Empty;
+    }
+
+    public IPropertySet GetProperties(string[] propertiesToReturn, string repositoryLocation)
+    {
+        var sourceControlProvider = new SourceControlProvider();
+        var provider = sourceControlProvider.GetProvider(repositoryLocation);
+
+        if (provider == null)
+        {
+            _log.Warning($"Path: {repositoryLocation} does not have an associated provider.");
+            return new PropertySet();
+        }
+
+        // This call does check the settings for file explorer and souce control integration.
+        return provider.GetProperties(propertiesToReturn, string.Empty);
+    }
+
+    private async Task<string> GetRepositoryLocationFromUser()
+    {
+        StorageFolder repositoryRootFolder = null;
+        try
+        {
+            using var folderDialog = new WindowOpenFolderDialog();
+            repositoryRootFolder = await folderDialog.ShowAsync(_window);
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Error occurred when selecting a folder for adding a repository.");
+            _telemetry.LogError(
+                ErrorEventName,
+                LogLevel.Critical,
+                new EnhanceRepositoryErrorEvent(nameof(SelectRepositoryAndMakeItEnhanced), ex.HResult, ex.Message, string.Empty));
+            return string.Empty;
+        }
+
+        if (repositoryRootFolder == null || repositoryRootFolder.Path.Length == 0)
+        {
+            _log.Information("Didn't select a location to register");
```

Comment:
Here and next log line, maybe specify "_user_ selected / didn't select" just so someone in the future doesn't read the log and wonder if it was the user or dev home.

Reply:
Done.